docs/environment_variables.md

```diff
@@ -139,18 +139,31 @@ Example:
 SSL_CERT_DIR=/path/to/ca-certs/ python -c "import httpx; httpx.get('https://example.com')"
 ```
 
-`HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY`
+`HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY`, `NO_PROXY`
```

Comment:
We might want to consider using a separate section for `NO_PROXY` (?), since it actually accepts a comma-separated list of URLs, and those URLs denote hosts, rather than proxies.

We have shared information about proxies in the existing section for now, but it's perfectly possible to move that as an introduction to a general "Proxies" section in this docs page.

So we can consider a structure like this…

```markdown
# Environment Variables

…

## Proxies

The environment variables documented below are used as a convention by various HTTP tooling, including:

* [cURL](https://github.com/curl/curl/blob/master/docs/MANUAL.md#environment-variables)
* [requests](https://github.com/psf/requests/blob/master/docs/user/advanced.rst#proxies)

For more information on using proxies in HTTPX, see [HTTP Proxying](/advanced/#http-proxying).

### `HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY`

Valid values: A URL to a proxy

...

### `NO_PROXY`

Valid values: a comma-separated list of hostnames
...
```

What do you think?

Reply:
By the way — I pushed https://github.com/encode/httpx/pull/1260 that we should review and merge before-hand so we can use the `###` style of headers on this page, like we do everywhere else. Happy to take reviews!

Reply:
Changed the doc according your comment.

Thank you for the advice about the splitting the paragraph into two